switchapi/es2k/lnw_v3/switch_pd_p4_name_routing.h

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2013-2021 Barefoot Networks, Inc.
+ * Copyright 2022-2023 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __SWITCH_PD_P4_NAME_ROUTING_MAPPING__
+#define __SWITCH_PD_P4_NAME_ROUTING_MAPPING__
+
+#ifdef __cplusplus
```

Comment:
1. The `extern "C"` isn't needed. This file defines preprocessor symbols, which are are the same for both C and C++.
2. The end-of-line comment on the #endif is unnecessary. We're only conditionalizing one line; there's no confusion over which condition we're closing; and extra text adds visual noise and makes the line within the conditional harder to read.

I've been removing the comment from the #endifs (and deleting the occasional superfluous `extern "C"`) as I work on files.